rllib/algorithms/bc/torch/bc_torch_learner.py

```diff
@@ -0,0 +1,60 @@
+import logging
+from typing import Mapping
+
+from ray.rllib.core.learner.learner import (
+    POLICY_LOSS_KEY,
+    LearnerHyperparameters,
+)
+from ray.rllib.core.learner.torch.torch_learner import TorchLearner
+from ray.rllib.core.rl_module.rl_module import ModuleID
+from ray.rllib.policy.sample_batch import SampleBatch
+from ray.rllib.utils.annotations import override
+from ray.rllib.utils.framework import try_import_torch
+from ray.rllib.utils.nested_dict import NestedDict
+from ray.rllib.utils.torch_utils import sequence_mask
+from ray.rllib.utils.typing import TensorType
+
+torch, nn = try_import_torch()
+logger = logging.getLogger(__file__)
+
+
+class BCTorchLearner(TorchLearner):
+    """Implements torch-specific BC loss logic."""
+
+    @override(TorchLearner)
+    def compute_loss_for_module(
+        self,
+        *,
+        module_id: ModuleID,
+        hps: LearnerHyperparameters,
+        batch: NestedDict,
+        fwd_out: Mapping[str, TensorType]
+    ) -> TensorType:
+        if self.module[module_id].is_stateful():
```

Comment:
Can we add a TODO here to fix this zero-padding requirement. What I have in mind is this:

If an algorithm supports RNNs (sequence-aware train batches), we should unify our format requirement and handling of these:
* The train batch should always have the format: (B, T, ...), so it has B rows, each of length T.
* If an episode ends in the middle of a row, we will simply insert a "initial state" at that position instead of doing zero-padding.
* A new field in the batch dict: "is_first" of shape (B, T) will indicate, at which exact B/T-positions to insert these initial states.
* Thus, we will NOT have to do any special reduction anymore. A simple torch.mean should suffice.

We don't have to fix this right now, but I do want to move to this pattern (which DreamerV3 already uses) in the mid-run. So far, this'll affect PPO and BC only.